### PR TITLE
Issue #3332: fix httplookup issue for get ns/topics in cpp

### LINKED ISSUE #3332: Cannot use topics regex pattern with using Go package (v2.2.1)
I'm trying to subscribe to multiple topics specified by regex pattern using following code:

```go
package main

import (
  "context"
  "fmt"

  "github.com/apache/incubator-pulsar/pulsar-client-go/pulsar"
)

func main() {
  client, err := pulsar.NewClient(pulsar.ClientOptions{
    URL: "http://127.0.0.1:8080",
  })
  if err != nil {
    panic(err)
  }
  defer client.Close()

  consumer, err := client.Subscribe(pulsar.ConsumerOptions{
    TopicsPattern:    "persistent://foo/bar/test-\\d\\d",
    SubscriptionName: "test",
    Type:             pulsar.Shared,
  })
  if err != nil {
    panic(err)
  }
  defer consumer.Close()

  for {
    cm, err := consumer.Receive(context.Background())
    if err != nil {
      panic(err)
    }

    fmt.Println(cm)
    consumer.Ack(cm)
  }
}
```

However the program immediately fails with:

```
INFO  | HTTPLookupService:222 | Curl Lookup Request sent for http://127.0.0.1:8080/admin/v2/namespaces/foo/bar/topics
INFO  | HTTPLookupService:236 | Response received for url http://127.0.0.1:8080/admin/v2/namespaces/foo/bar/topics code 200
terminate called after throwing an instance of 'Json::LogicError'
  what():  in Json::Value::resolveReference(key, end): requires objectValue
signal: aborted (core dumped)
```

It seems that it successfully retrieves the list of topics for tenant/namespace but fails to parse the response. When curling the endpoint I get:

```
$ curl -L 'http://127.0.0.1:8080/admin/v2/namespaces/foo/bar/topics'
["persistent://foo/bar/test-aa","persistent://foo/bar/test-ab"]
```

This is not what the undelying C++ library expects as seen [here](https://github.com/apache/pulsar/blob/v2.2.1/pulsar-client-cpp/lib/HTTPLookupService.cc#L319). It seems that it looks for a document with key `topics` in it but all it gets is JSON array.

### PR DESCRIPTION
**Pull request by jiazhai**
Fixes #3332

### Motivation

In http request to get ns/topics, it will return an array of topics,
```
$ curl -L 'http://127.0.0.1:8080/admin/v2/namespaces/foo/bar/topics'
["persistent://foo/bar/test-aa","persistent://foo/bar/test-ab"]
```
But in cpp method `HTTPLookupService::parseNamespaceTopicsData`, it is expecting a json of item 
like: `{"topics": ["topic1", "topic2"] }`. This may be caused by the response format changes.

### Modifications

- change cpp method `HTTPLookupService::parseNamespaceTopicsData` to following current http response data format.
- changed cpp test `testPatternMultiTopicsConsumerPubSub` to use http adminUrl, to protect this part of cpp code.

### Verifying this change
cpp unit tests passed, especially for test `testPatternMultiTopicsConsumerPubSub`

### CODE REVIEW

**Comment by jiazhai** [comment]
rerun cpp tests

File: pulsar-client-cpp/lib/HTTPLookupService.cc [hunk 1]
```diff
@@ -310,18 +310,21 @@ LookupDataResultPtr HTTPLookupService::parseLookupData(const std::string &json)
 NamespaceTopicsPtr HTTPLookupService::parseNamespaceTopicsData(const std::string &json) {
     Json::Value root;
     Json::Reader reader;
+    LOG_INFO("GetNamespaceTopics json = " << json);
```
Comment (merlimat) [inline pulsar-client-cpp/lib/HTTPLookupService.cc]: This should be at debug level

File: pulsar-client-cpp/tests/BasicEndToEndTest.cc [hunk 2]
```diff
@@ -1756,7 +1756,7 @@ TEST(BasicEndToEndTest, testPatternTopicsConsumerInvalid) {
 // verify PatternMultiTopicsConsumer subscribed matched topics,
 // and only receive messages from matched topics.
 TEST(BasicEndToEndTest, testPatternMultiTopicsConsumerPubSub) {
-    Client client(lookupUrl);
+    Client client(adminUrl);
```
Comment (merlimat) [inline pulsar-client-cpp/tests/BasicEndToEndTest.cc]: Ideally we should have the test fetching the lists of topic using both HTTP and protobuf.

File: pulsar-client-cpp/lib/HTTPLookupService.cc (hunk 1 above)
Comment (sijie) [inline pulsar-client-cpp/lib/HTTPLookupService.cc]: nit: debug

**Comment by jiazhai** [comment]
rerun integration tests

**Comment by jiazhai** [comment]
rerun integration tests

**Comment by jiazhai** [comment]
rerun integration tests

**Pull request merged**